AnkiDroid/src/main/java/com/ichi2/anki/AddonsBrowser.java

```diff
@@ -0,0 +1,438 @@
+                npmAddonName = npmAddonName.replaceAll("\u00A0", "");
+
+                UIUtils.showThemedToast(this, getString(R.string.downloading_addon), true);
+                AddonsNpmUtility.getPackageJson(npmAddonName, this);
+
+                downloadDialog.dismiss();
+
+            });
+
+            downloadDialog.show();
+            return true;
+        });
+
+        mGetMoreAddons.setOnMenuItemClickListener(item -> {
+            openUrl(Uri.parse(getResources().getString(R.string.ankidroid_js_addon_npm_search_url)));
+            return true;
+        });
+
+        return super.onCreateOptionsMenu(menu);
+    }
+
+
+    /*
+       list addons with valid package.json, i.e contains
+       ankidroid_js_api = 0.0.1
+       keywords='ankidroid-js-addon'
+       and non empty string.
+       Then that addon will available for enable/disable
+     */
+    public void listAddonsFromDir(String addonType) {
+        boolean success = true;
+        if (!addonsHomeDir.exists()) {
+            success = addonsHomeDir.mkdirs();
+        }
+
+        if (success) {
+
+            // reset for new view create
+            addonsNames.clear();
+
+            File directory = new File(String.valueOf(addonsHomeDir));
+            File[] files = directory.listFiles();
+            for (File file : files) {
+                Timber.d("Addons:%s", file.getName());
+
+                // Read package.json from
+                // AnkiDroid/addons/ankidroid-addon-../package/package.json
+                File addonsPackageDir = new File(file, "package");
+                File addonsPackageJson = new File(addonsPackageDir, "package.json");
+
+                JSONObject jsonObject = AddonsNpmUtility.packageJsonReader(addonsPackageJson);
+
+                AddonModel addonModel;
+                if (isValidAddonPackage(jsonObject)) {
+
+                    String addonName = jsonObject.optString("name", "");
+                    String addonVersion = jsonObject.optString("version", "");
+                    String addonDev = jsonObject.optString("author", "");
+                    String addonAnkiDroidAPI = jsonObject.optString("ankidroid_js_api", "");
+                    String addonHomepage = jsonObject.optString("homepage", "");
+                    String typeOfAddon = jsonObject.optString("addon_type", "");
+
+                    if (addonType.equals(typeOfAddon)) {
+                        addonModel = new AddonModel(addonName, addonVersion, addonDev, addonAnkiDroidAPI, addonHomepage, addonType);
+                        addonsNames.add(addonModel);
+                    }
+                }
+            }
+
+            addonsListRecyclerView.setAdapter(new AddonsAdapter(addonsNames, this));
+
+            if (addonsNames.isEmpty()) {
+                addonsMessageLayout.setVisibility(View.VISIBLE);
+            } else {
+                addonsMessageLayout.setVisibility(View.GONE);
+            }
```

Comment:
I don't know what this does without context (what is `addonsMessageLayout`?)

Consider:
* Including a comment as to why this happens, 
* Extract to a named method where the method name gives context
* Renaming the variable to be more descriptive

```suggestion
            addonsMessageLayout.setVisibility(addonsNames.isEmpty() ? View.VISIBLE : View.GONE);
```

Reply:
This is extension icon with message below on Addons Browser first screen when no addons available.